core_backend/app/users/models.py

```diff
@@ -0,0 +1,88 @@
+from datetime import datetime
+from typing import Optional
+
+from sqlalchemy import (
+    DateTime,
+    String,
+    # delete,
+    select,
+)
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import Mapped, mapped_column
+
+from ..models import Base
+from .schemas import UserCreate
+
+
+class UserDB(Base):
+    """
+    SQL Alchemy data model for users
+    """
+
+    __tablename__ = "user"
+
+    user_id: Mapped[str] = mapped_column(String, primary_key=True, nullable=False)
+    username: Mapped[str] = mapped_column(String, nullable=False)
+
+    # retrieval_token: Mapped[str] = mapped_column(String, nullable=False)
+
+    created_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
+    updated_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
+
+    def __repr__(self) -> str:
+        """Pretty Print"""
+        return f"<{self.username} mapped to #{self.user_id}>"
+
+
+async def save_user_to_db(
+    user: UserCreate,
+    asession: AsyncSession,
+) -> UserDB:
+    """
+    Saves a user in the database
+    """
+
+    # Check if user with same username already exists
+    stmt = select(UserDB).where(UserDB.username == user.username)
+    result = await asession.execute(stmt)
+    try:
+        result.scalar_one()
```

Comment:
/nit should this be just `.one()`. Are we adding overhead by asking it to be `scalar` or does it make more efficient?

Reply:
This was a copilot suggestion. [.scalar_one()](https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.Result.scalar_one) does two operations - we just need [.one()](https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.Result.one),  thanks for picking this up. I'll also add an exception for [MultipleResultsFound](https://docs.sqlalchemy.org/en/14/core/exceptions.html#sqlalchemy.exc.MultipleResultsFound).